irohad/ametsuchi/mutable_storage.hpp

```diff
@@ -42,37 +31,34 @@ namespace iroha {
      public:
       /**
        * Predicate type checking block
+       * Function parameters:
+       *  - Block - block to be checked
+       *  - PeerQuery - interface for ledger peers list retrieval
+       *  - HashType - hash of top block in blockchain
        */
       using MutableStoragePredicate =
           std::function<bool(const shared_model::interface::Block &,
                              PeerQuery &,
                              const shared_model::interface::types::HashType &)>;
 
       /**
-       * Checks if block satisfies predicated
-       * @param block block to be checked
-       * @param predicate function returning true if predicate satisfied and
-       * false otherwise
-       * @return result of predicate
+       * Applies block without additional validation function
+       * @see apply(block, function)
        */
-      virtual bool check(const shared_model::interface::Block &block,
-                         MutableStoragePredicate function) = 0;
+      virtual bool apply(const shared_model::interface::Block &block) = 0;
 
       /**
-       * Applies a block to current mutable state
-       * using logic specified in function
-       * @param block Block to be applied
+       * Applies an observable of blocks to current mutable state using logic
+       * specified in function
+       * @param blocks Blocks to be applied
        * @param function Function that specifies the logic used to apply the
```

Comment:
But the `function` does not provide logic for block application, it's still a predicate?